src/Samples/Sarif.WorkItems.Sample/ProgramStubs.cs

```diff
@@ -0,0 +1,233 @@
+        }
+
+        public static string GetResourceText(string resourceName)
+        {
+            string text = null;
+
+            using (Stream stream = typeof(Program).Assembly.GetManifestResourceStream(resourceName))
+            {
+                if (stream == null) { return null; }
+
+                using (StreamReader reader = new StreamReader(stream))
+                {
+                    text = reader.ReadToEnd();
+                }
+            }
+            return text;
+        }
+
+        private interface ISarifLogBaseliner
+        {
+            SarifLog CreateBaselinedSarifLog(SarifLog baselineSarif, SarifLog currentSarif, bool assignGuidsToNewResults);
+            SarifLog CreateBaselinedSarifLog(SarifLog baselineSarif, SarifLog currentSarif);
+        }
+
+        private class SarifLog
+        {
+            public IList<Run> Runs { get; internal set; }
+
+            internal static IEnumerable<WorkItemFilingMetadata> CreateWorkItemFilingMetadataFromLogs(IEnumerable<SarifLog> sarifLogsToFile)
+            {
+                throw new NotImplementedException();
+            }
+
+            public class Run
+            {
+                public IList<Result> Results { get; internal set; }
+            }
+        }
+
+        private class SarifLogBaselinerFactory
+        {
+            internal static ISarifLogBaseliner CreateSarifLogBaseliner(object standard)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class SarifBaselineType
+        {
+            public static object Standard { get; internal set; }
+        }
+
+        private class SarifSplitter
+        {
+            internal static IEnumerable<SarifLog> Split(SarifLog sarifLog, Func<Result, bool> filterPredicate, SplittingStrategy splittingStrategy)
+            {
+                throw new NotImplementedException();
+            }
+
+            internal class Result
+            {
+                public BaselineState BaselineState;
+            }
+
+            internal class SarifSplittingStrategy
+            {
+            }
+        }
+
+        private enum BaselineState
+        {
+            New
+        }
+
+        private enum SplittingStrategy
+        {
+            OneFilePerRunPerRule
+        }
+
+        private class WorkItemFilingMetadata
+        {
+            public string Title { get; internal set; }
+
+            public string AreaPath { get; internal set; }
+
+            public string Body { get; internal set; }
+
+            public HashSet<string> Tags { get; internal set; }
+
+            public Uri Uri { get; internal set; }
+
+            public List<object> Attachments { get; internal set; }
+        }
+
+
+        private static void LogFailure(Exception exception)
+        {
+            throw new NotImplementedException();
+        }
+
+        private class AdoWorkItemFiler
```

Comment:
>AdoWorkItemFiler [](start = 22, length = 16)

See my comments in the program -- I'd prefer a plain `WorkItemFiler` with an injected `WorkItemHost`-derived object (`new AdoWorkItemHost()` or `new GitHubWorkItemHost()`.

Reply:
Yes, there should be a vanilla work item filer, but we should also create an instance of this that injected the standard AdoWorkItemHost(). This is more usable for API consumers (again, this is a top-down exercise of trying to get the API use case in the best form for consumers). 

---
In reply to: [307088864](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307088864) [](ancestors = 307088864)